core/filtermaps/filtermaps.go

```diff
@@ -358,38 +376,37 @@ func (f *FilterMaps) init() error {
 
 // removeBloomBits removes old bloom bits data from the database.
 func (f *FilterMaps) removeBloomBits() {
-	f.safeDeleteRange(rawdb.DeleteBloomBitsDb, "Removing old bloom bits database")
+	f.safeDeleteWithLogs(rawdb.DeleteBloomBitsDb, "Removing old bloom bits database", f.isShuttingDown)
 	f.closeWg.Done()
 }
 
-// safeDeleteRange calls the specified database range deleter function
-// repeatedly as long as it returns leveldb.ErrTooManyKeys.
-// This wrapper is necessary because of the leveldb fallback implementation
-// of DeleteRange.
-func (f *FilterMaps) safeDeleteRange(removeFn func(ethdb.KeyValueRangeDeleter) error, action string) bool {
-	start := time.Now()
-	var retry bool
-	for {
-		err := removeFn(f.db)
-		if err == nil {
-			if retry {
-				log.Info(action+" finished", "elapsed", time.Since(start))
-			}
-			return true
-		}
-		if err != leveldb.ErrTooManyKeys {
-			log.Error(action+" failed", "error", err)
-			return false
-		}
-		select {
-		case <-f.closeCh:
-			return false
-		default:
-		}
-		if !retry {
+// safeDeleteWithLogs is a wrapper for a function that performs a safe range
+// delete operation using rawdb.SafeDeleteRange. It emits log messages if the
+// process takes long enough to call the stop callback.
+func (f *FilterMaps) safeDeleteWithLogs(deleteFn func(db ethdb.KeyValueStore, hashScheme bool, stopCb func() bool) error, action string, stopCb func() bool) error {
+	var (
+		start          = time.Now()
+		logPrinted     bool
+		lastLogPrinted = start
+	)
+	switch err := deleteFn(f.db, f.hashScheme, func() bool {
```

Comment:
It's very weird to print logs if there are nothing to delete,

e.g. 

```
INFO [03-31|15:07:25.452] Removing old bloom bits database in progress... elapsed=17.430236ms
INFO [03-31|15:07:25.644] Removing old bloom bits database finished elapsed=209.842669ms
```
\
These two logs are printed even the legacy bloom data are not existent

Reply:
Fixed